test/controllers/GoodsReleasedControllerSpec.scala

```diff
@@ -29,63 +29,76 @@ import models.State
 import models.request.ArrivalId
 import org.mockito.Matchers.{eq => eqTo, _}
 import org.mockito.Mockito._
+import org.scalatest.BeforeAndAfterEach
 import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
 import play.api.inject.bind
 import play.api.test.FakeRequest
 import play.api.test.Helpers._
 import repositories.ArrivalMovementRepository
 import repositories.LockRepository
+import services.XmlValidationService
+import uk.gov.hmrc.http.BadRequestException
 import utils.Format
 
 import scala.concurrent.Future
 import scala.util.Failure
 import scala.util.Success
 
-class GoodsReleasedControllerSpec extends SpecBase with ScalaCheckPropertyChecks with ModelGenerators {
+class GoodsReleasedControllerSpec extends SpecBase with ScalaCheckPropertyChecks with ModelGenerators with BeforeAndAfterEach {
+
+  private val mockArrivalMovementRepository: ArrivalMovementRepository = mock[ArrivalMovementRepository]
+  private val mockLockRepository: LockRepository                       = mock[LockRepository]
+  private val mockXmlValidationService: XmlValidationService           = mock[XmlValidationService]
+
+  private val dateOfPrep = LocalDate.now()
+  private val timeOfPrep = LocalTime.of(1, 1)
+
+  private val arrivalId     = ArrivalId(1)
+  private val version       = 1
+  private val messageSender = MessageSender(arrivalId, version)
+  private val arrival = Arrival(
+    arrivalId,
+    MovementReferenceNumber("mrn"),
+    "eori",
+    State.Submitted,
+    ArrivalDateTime(dateOfPrep, timeOfPrep),
+    ArrivalDateTime(dateOfPrep, timeOfPrep),
+    Seq.empty
+  )
+
+  private val requestXmlBody =
+    <CC025A>
+      <DatOfPreMES9>{Format.dateFormatted(dateOfPrep)}</DatOfPreMES9>
+      <TimOfPreMES10>{Format.timeFormatted(timeOfPrep)}</TimOfPreMES10>
+    </CC025A>
+
+  override def beforeEach: Unit = {
+    super.beforeEach()
+    reset(mockArrivalMovementRepository)
+    reset(mockLockRepository)
+    reset(mockXmlValidationService)
+  }
```

Comment:
I'm not sure I see the benefit of doing this? All we've done is create global mutable state that can leak between tests and we've been forced to bring in test hooks to clean up that global mutable data.